walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,95 @@
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
+    }
+    // If there are no events, then return the whole day.
+    if (events.isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    // variables needed throughout the function
+    Collection<String> request_attendees_list = request.getAttendees();
+    Collection<Event> event_list = new ArrayList<Event>();
+    // Loops through the events parameter to add to an event_list collection
+    // if the request attendees is related to the attendees of the event. If
+    // the event attendees do not match the request attendees, then the event
+    // is ignored since it is not related to the request. 
+    for (Event e: events) {
+        for (String r : request_attendees_list) {
+             if (e.getAttendees().contains(r)) {
+                 event_list.add(e);
+                 break;
+             }
+        }
+    }
+    // After looping for related events, if there is none, then return the whole day. 
+    if (event_list.size() == 0) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    // If there is a single related event, return the range from before and after the event. 
+    if (event_list.size() == 1) {
+        List<TimeRange> single_time_range_list = new ArrayList<TimeRange>();
+        for (Event e : event_list) {
+            single_time_range_list.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, e.getWhen().start(), false));
+            single_time_range_list.add(TimeRange.fromStartEnd(e.getWhen().end(), TimeRange.END_OF_DAY, true));
+        }
+        return single_time_range_list;
+    }
+    // If there are more than one related event, then follow more specific steps below. 
+    if (events.size() > 1) {
+        List<TimeRange> multiple_time_range_list = new ArrayList<TimeRange>();
+        List<Integer> start_time_list = new ArrayList<Integer>();
+        List<Integer> end_time_list = new ArrayList<Integer>();
+        // Add the start and end times for each event into a list. 
+        for (Event e : event_list) {
+            start_time_list.add(e.getWhen().start());
+            end_time_list.add(e.getWhen().end());
+        }
+        // If the starting time is not 0, then create the first time range from
+        // the morning to the first element of start_time_list. 
+        if (start_time_list.get(0) != 0) {
+            multiple_time_range_list.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, start_time_list.get(0), false));
+        }
+        // Loop to check to if the first element in end_time_list at is less 
+        // than the start_time_list at i+1. If the statement is true, then
+        // a time range is added ranging from the element of end_time_list at 
+        // i to the start_time i+1. Otherwise, the for loop skips it. This
+        // loop checks until the second to last element in end_time_list. 
+        for (int i = 0; i < end_time_list.size()-1; i++) {
```

Comment:
This works, but we also have some existing function that you might want to check out: https://github.com/KaylaNguyen/SPS/blob/master/walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/TimeRange.java#L82